### PR TITLE
log how long GitHub expect us to wait for retry

### PR DESCRIPTION
**Pull request by superyyrrzz**
#4139

### CODE REVIEW

File: src/docfx/lib/github/GitHubAccessor.cs [hunk 1]
```diff
@@ -60,6 +60,10 @@ public async Task<(Error, GitHubUser)> GetUserByLogin(string login)
                 _rateLimitError = Errors.GitHubApiFailed(apiDetail, ex);
                 return (_rateLimitError, null);
             }
+            catch (AbuseException ex)
+            {
+                return (Errors.GitHubApiFailed(apiDetail, ex, ex.RetryAfterSeconds), null);
```
Comment (OsmondJiang) [inline src/docfx/lib/github/GitHubAccessor.cs]: can we combine these two api exception handler together? seems they are almost same

File: src/docfx/Errors.cs [hunk 2]
```diff
@@ -86,8 +86,15 @@ public static Error GitHubUserNotFound(string login)
         ///   - the api call reach github limit
         ///   - using invalid access token(more detailed info in ex.Message)
         /// </summary>
-        public static Error GitHubApiFailed(string api, Exception ex)
-            => new Error(ErrorLevel.Warning, "github-api-failed", $"Failed calling GitHub API '{api}': {ex.Message}");
+        public static Error GitHubApiFailed(string api, Exception ex, int? retryAfterSeconds = null)
```
Comment (yufeih) [inline src/docfx/Errors.cs]: Don't use error messages to log diagnostics, if you want to log just `Log.Write()`

File: src/docfx/lib/github/GitHubAccessor.cs [hunk 3]
```diff
@@ -102,6 +102,10 @@ public async Task<(Error, string login)> GetLoginByCommit(string repoOwner, stri
             }
             catch (Exception ex)
             {
+                if (ex is AbuseException aex)
+                {
+                    Log.Write($"Failed calling GitHub API '{apiDetail}', message: '{ex.Message}', retryAfterSeconds: '{aex.RetryAfterSeconds}'");
```
Comment (OsmondJiang) [inline src/docfx/lib/github/GitHubAccessor.cs]: only applied to GetLoginByCommit? What about GetUserByLogin?

Comment (superyyrrzz) [reply src/docfx/lib/github/GitHubAccessor.cs]: Forgot about this... added

**Pull request merged**